monix-reactive/shared/src/main/scala/monix/reactive/internal/builders/IntervalFixedDelayObservable.scala

```diff
@@ -63,11 +63,7 @@ private[reactive] final class IntervalFixedDelayObservable(initialDelay: FiniteD
       }
     }
 
-    if (initialDelay.length <= 0)
-      runnable.run()
-    else
-      task := s.scheduleOnce(initialDelay.length, initialDelay.unit, runnable)
-
+    task := s.scheduleOnce(initialDelay.length, initialDelay.unit, runnable)
```

Comment:
I believe we could do a small optimization and call

`s.execute(runnable)` in case of `initialDelay.length <= 0`.

It makes the following test fail:

```scala
  test("issue #468 - concat is cancellable") { implicit sc =>
    var items = 0

    val a = Observable.now(1L)
    val b = Observable.interval(1.second)
    val c = (a ++ b).doOnNextF(_ => Task { items += 1 })
    val d = c.take(10).subscribe()

    assert(items > 0, "items > 0")
    assert(sc.state.tasks.nonEmpty, "tasks.nonEmpty")

    d.cancel()
    assert(sc.state.tasks.isEmpty, "tasks.isEmpty")
  }
```

But I believe it is because the test needs to be adjusted.

Before the change it processes `a` then immediately the first element from `b`
Now it processes `a` then schedules `b` which is waiting to be executed so it's stuck in tasks queue.
In both cases the `b` is cancelled.

I think in this case it would be OK to do `s.tick(); d.cancel()` in the test

WDYT? Any thoughts @alexandru ?

Reply:
> s.execute(runnable) in case of initialDelay.length <= 0.

I think that optimization should be done in the `Scheduler` implementations, not here. We could double check if it is, but I remember the optimization is being applied.

Reply:
Well, doing the optimization here as well is fine too, I personally don't mind it.